analysis/parser/RawTextParser/src/main/kotlin/de/maibornwolff/codecharta/parser/rawtextparser/ProjectGenerator.kt

```diff
@@ -8,9 +8,9 @@ import de.maibornwolff.codecharta.model.ProjectBuilder
 
 class ProjectGenerator(private val projectBuilder: ProjectBuilder = ProjectBuilder()) {
 
-    fun generate(projectMetrics: ProjectMetrics, pipedProject: Project?): Project {
+    fun generate(projectMetrics: ProjectMetrics, maxIndentLevel: Int, pipedProject: Project?): Project {
```

Comment:
I think the maxIndentLevel should have a default value here so the function can be called without it

Reply:
The ProjectGenerator is only called once within the RawTextParser, with the maximum indentation level set. - there's no need to make this an optional parameter.